woocommerce/woocommerce-android
# Issue/2708 Display downloadable files in the product details

Fixes #2708

**Summary**
This PR allows displaying the downloadable files with their details on the product details fragment.

**Changes**

- Added a feature flag M4 to enable the feature
- Updated FluxC version to use the changes done on PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1663
- Updated the card of downloadable files in the list of product details, when M4 feature is enabled, the old card won't be shown too.
- Added FragmentProductDownloads to display the list of files.
- Fixed the test `Displays the product detail properties correctly`, and updated it to confirm the handling of downloadable files correctly.

**Screenshots**
![Screen Shot 2020-08-13 at 19 39 11](https://user-images.githubusercontent.com/1657201/90173663-b6d0df00-dd9c-11ea-9ec3-efd1810853d1.png) ![Screen Shot 2020-08-13 at 19 39 20](https://user-images.githubusercontent.com/1657201/90173669-b89aa280-dd9c-11ea-9580-8bbdb7ff3002.png)

**Testing**
- Make sure that one of the products in the store is downloadable, and has some files added in wp-admin.
- Go to the product details.
- Notice the downloadable files section.
- Click on the downloadable files section to check the details of each file.
